### PR TITLE
MM-38731 - Fix for: Old status update dialog partially exposed in web/desktop

### PR DESCRIPTION
**Pull request by cpoile**
#### Summary
- Trigger the new style dialog using a custom post - it works for webapp and automatically falls back to the interactive dialog for mobile

#### Ticket Link
- https://mattermost.atlassian.net/browse/MM-38731

#### Checklist

- [ ] ~~Telemetry updated~~
- [ ] ~~Gated by experimental feature flag~~
- [ ] ~~Unit tests updated~~

### CODE REVIEW

File: server/api/playbook_runs.go [hunk 1]
```diff
@@ -801,23 +802,54 @@ func (h *PlaybookRunHandler) reminderButtonUpdate(w http.ResponseWriter, r *http
...
 		return
 	}
 
-	playbookRunID, err := h.playbookRunService.GetPlaybookRunIDForChannel(requestData.ChannelId)
+	h.removeReminderPost(w, userID, playbookRunID, requestData.ChannelId)
+
+	ReturnJSON(w, nil, http.StatusOK)
+}
+
+func (h *PlaybookRunHandler) removeReminderPost(w http.ResponseWriter, userID, playbookID, channelID string) {
```
Comment (agarciamontoro) [inline server/api/playbook_runs.go:839]: I think this is a typo, right?
```suggestion
func (h *PlaybookRunHandler) removeReminderPost(w http.ResponseWriter, userID, playbookRunID, channelID string) {
```

File: webapp/src/components/update_request_post.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    post: Post;
+}
+
+export const UpdateRequestPost = (props: Props) => {
+    const dispatch = useDispatch();
+    const channel = useSelector<GlobalState, Channel>((state) => getChannel(state, props.post.channel_id));
+    const team = useSelector<GlobalState, Team>((state) => getTeam(state, channel.team_id));
+    const currentRun = useSelector(currentPlaybookRun);
+    const targetUsername = props.post.props.targetUsername ?? '';
+    const playbookId = currentRun?.id;
```
Comment (agarciamontoro) [inline webapp/src/components/update_request_post.tsx:32]: Same typo here:
```suggestion
    const playbookRunId = currentRun?.id;
```

**Review by agarciamontoro — CHANGES_REQUESTED** [review]
Nice and clean! Just a couple of naming typos that we should fix to not confuse our future selves.

File: server/api/playbook_runs.go (hunk 1 above)
Comment (cpoile) [reply server/api/playbook_runs.go:839]: Ahhh, it strikes again! Thanks for catching that.

File: webapp/src/components/update_request_post.tsx (hunk 2 above)
Comment (cpoile) [reply webapp/src/components/update_request_post.tsx:32]: 👍

**Comment by cpoile** [comment]
@abhijit-singh Does this look good for replacing the interactive dialog (when the user is using the webapp -- ie, not mobile):

image

**Review by agarciamontoro — APPROVED** [review]
:tada:

**Review by crspeller — APPROVED** [review]

**Comment by abhijit-singh** [comment]
Thanks @cpoile! A couple of suggested changes:

* Can we wrap the buttons in a box similar to the one we have for status updates?
image

* I think the buttons in the post can always have the standard blue color instead of red, as in the screenshot above.

**Review by abhijit-singh — CHANGES_REQUESTED** [review]
Thanks @cpoile! A couple of suggested changes:

* Can we wrap the buttons in a container similar to what we have for the status update post?
image

* I think the buttons in the post can always be of the standard blue color (instead of red), as in the screenshot above.

**Comment by cpoile** [comment]
@abhijit-singh Look good?
image

**Review by abhijit-singh — APPROVED** [review]

**Pull request merged**